File: bamboo_/ZATools/DYreweighting/perceptron_nn/dyrwt_fct_boosted_MuMu_2017.cxx
#include "dyrwt_fct_boosted_MuMu_2017.h"
#include <cmath>

double dyrwt_fct_boosted_MuMu_2017::Value(int index,double in0) {
   input0 = (in0 - 0)/1;
   switch(index) {
     case 0:
         return neuron0x947d0d0();
     default:
         return 0.;
   }
}

double dyrwt_fct_boosted_MuMu_2017::Value(int index, double* input) {
   input0 = (input[0] - 0)/1;
   switch(index) {
     case 0:
         return neuron0x947d0d0();
     default:
         return 0.;
   }
}

double dyrwt_fct_boosted_MuMu_2017::neuron0x947cd90() {
   return input0;
}

double dyrwt_fct_boosted_MuMu_2017::input0xa45d300() {
   double input = -21.7169;
   input += synapse0x832ae10();
   return input;
}

double dyrwt_fct_boosted_MuMu_2017::neuron0xa45d300() {
   double input = input0xa45d300();
   return ((input < -709. ? 0. : (1/(1+exp(-input)))) * 1)+0;
}

double dyrwt_fct_boosted_MuMu_2017::input0xa45d5b0() {
   double input = 1.0068;
   input += synapse0x9320010();
   return input;
}

double dyrwt_fct_boosted_MuMu_2017::neuron0xa45d5b0() {
   double input = input0xa45d5b0();
   return ((input < -709. ? 0. : (1/(1+exp(-input)))) * 1)+0;
}

double dyrwt_fct_boosted_MuMu_2017::input0xa45d8f0() {
   double input = 2.3476;
   input += synapse0xa464e60();
   return input;
}

double dyrwt_fct_boosted_MuMu_2017::neuron0xa45d8f0() {
   double input = input0xa45d8f0();
   return ((input < -709. ? 0. : (1/(1+exp(-input)))) * 1)+0;
}

double dyrwt_fct_boosted_MuMu_2017::input0xa45dcd0() {
   double input = -6.29544;
   input += synapse0x94a5a90();
   return input;
}

double dyrwt_fct_boosted_MuMu_2017::neuron0xa45dcd0() {
   double input = input0xa45dcd0();
   return ((input < -709. ? 0. : (1/(1+exp(-input)))) * 1)+0;
}

double dyrwt_fct_boosted_MuMu_2017::input0x947d0d0() {
   double input = 1.07758;
   input += synapse0xa45e130();
   input += synapse0xa45e170();
   input += synapse0xa45e1b0();
   input += synapse0xa45e1f0();
   return input;
}

double dyrwt_fct_boosted_MuMu_2017::neuron0x947d0d0() {
   double input = input0x947d0d0();
   return (input * 1)+0;
}

double dyrwt_fct_boosted_MuMu_2017::synapse0x832ae10() {
   return (neuron0x947cd90()*0.0226446);
}

double dyrwt_fct_boosted_MuMu_2017::synapse0x9320010() {
   return (neuron0x947cd90()*-0.843391);
}

double dyrwt_fct_boosted_MuMu_2017::synapse0xa464e60() {
   return (neuron0x947cd90()*-0.301418);
}

double dyrwt_fct_boosted_MuMu_2017::synapse0x94a5a90() {
   return (neuron0x947cd90()*0.0109543);
}

double dyrwt_fct_boosted_MuMu_2017::synapse0xa45e130() {
   return (neuron0xa45d300()*-24.4656);
}

double dyrwt_fct_boosted_MuMu_2017::synapse0xa45e170() {
   return (neuron0xa45d5b0()*1.19219);
}

double dyrwt_fct_boosted_MuMu_2017::synapse0xa45e1b0() {
   return (neuron0xa45d8f0()*-2.27515);
}

double dyrwt_fct_boosted_MuMu_2017::synapse0xa45e1f0() {
   return (neuron0xa45dcd0()*4.12597);
}

File: bamboo_/data/DYreweighting_run2/perceptron_nn/dyrwt_fct_boosted_MuMu_2017.py
from math import exp

from math import tanh

class dyrwt_fct_boosted_MuMu_2017:
	def value(self,index,in0):
		self.input0 = (in0 - 0)/1
		if index==0: return self.neuron0x947d0d0();
		return 0.
	def neuron0x947cd90(self):
		return self.input0
	def neuron0xa45d300(self):
		input = -21.7169
		input = input + self.synapse0x832ae10()
		if input<-709. : return 0
		return ((1/(1+exp(-input)))*1)+0
	def neuron0xa45d5b0(self):
		input = 1.0068
		input = input + self.synapse0x9320010()
		if input<-709. : return 0
		return ((1/(1+exp(-input)))*1)+0
	def neuron0xa45d8f0(self):
		input = 2.3476
		input = input + self.synapse0xa464e60()
		if input<-709. : return 0
		return ((1/(1+exp(-input)))*1)+0
	def neuron0xa45dcd0(self):
		input = -6.29544
		input = input + self.synapse0x94a5a90()
		if input<-709. : return 0
		return ((1/(1+exp(-input)))*1)+0
	def neuron0x947d0d0(self):
		input = 1.07758
		input = input + self.synapse0xa45e130()
		input = input + self.synapse0xa45e170()
		input = input + self.synapse0xa45e1b0()
		input = input + self.synapse0xa45e1f0()
		return (input*1)+0
	def synapse0x832ae10(self):
		return (self.neuron0x947cd90()*0.0226446)
	def synapse0x9320010(self):
		return (self.neuron0x947cd90()*-0.843391)
	def synapse0xa464e60(self):
		return (self.neuron0x947cd90()*-0.301418)
	def synapse0x94a5a90(self):
		return (self.neuron0x947cd90()*0.0109543)
	def synapse0xa45e130(self):
		return (self.neuron0xa45d300()*-24.4656)
	def synapse0xa45e170(self):
		return (self.neuron0xa45d5b0()*1.19219)
	def synapse0xa45e1b0(self):
		return (self.neuron0xa45d8f0()*-2.27515)
	def synapse0xa45e1f0(self):
		return (self.neuron0xa45dcd0()*4.12597)
